Content.Server/GameObjects/Components/SingularityComponent.cs

```diff
@@ -0,0 +1,163 @@
+{
+    [RegisterComponent]
+    public class SingularityComponent : Component, ICollideBehavior
+    {
+        public override string Name => "Singularity";
+
+        public int Energy = 100;
+        public int Level = 1;
+
+        private Random rand = new Random();
+
+        private SingularityController _singularityController;
+        private IEntityManager _entityManager;
+
+        private ICollidableComponent _collidableComponent;
+
+        private IMapManager _mapManager;
+
+        private SpriteComponent _spriteComponent;
+
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            _collidableComponent = Owner.GetComponent<ICollidableComponent>();
+            _collidableComponent.Hard = false;
+
+            _spriteComponent = Owner.GetComponent<SpriteComponent>();
+
+            _singularityController = _collidableComponent.EnsureController<SingularityController>();
+            _singularityController.ControlledComponent = _collidableComponent;
+
+            _entityManager = IoCManager.Resolve<IEntityManager>();
+            _mapManager = IoCManager.Resolve<IMapManager>();
+
+
+        }
+
+        public void Update()
+        {
+            Energy--;
+
+            _singularityController.Push(new Vector2((rand.Next(-10, 10)), rand.Next(-10, 10)).Normalized, 5f);
+
+            foreach (var entity in _entityManager.GetEntitiesInRange(Owner.Transform.GridPosition, 15))
+            {
+                if (entity.HasComponent<RadiationPanel>())
+                {
+                    entity.GetComponent<RadiationPanel>().Radiation += Level * 100;
+                }
+            }
+
+            UpdateLevel();
+        }
+
+        public void TileUpdate()
+        {
+            IMapGrid mapGrid = _mapManager.GetGrid(Owner.Transform.GridID);
+            foreach (TileRef tile in mapGrid.GetTilesIntersecting(_collidableComponent.WorldAABB))
+            {
+                mapGrid.SetTile(tile.GridIndices, Tile.Empty);
+            }
+        }
+
+        public void UpdateLevel()
+        {
+            int prevLevel = Level;
+            float radius = 0.5f;
+
+            if (Energy == 0)
+            {
+                //collapse
+            }
+
+            if (Energy < 200)
+            {
+                Level = 1;
+            }
+
+            if (Energy >= 200)
+            {
+                Level = 2;
+                radius = 1.5f;
+            }
+
+            if (Energy >= 300)
+            {
+                Level = 3;
+                radius = 2.5f;
+            }
+
+            if (Energy >= 600)
+            {
+                Level = 4;
+                radius = 3.5f;
+            }
+
+            if (Energy >= 1000)
+            {
+                Level = 5;
+                radius = 4.5f;
+            }
+
+            if (Energy >= 1500)
+            {
+                Level = 6;
+                radius = 5.5f;
+            }
```

Comment:
```suggestion
            if(Energy == 0)
            {
                //Collapse...
            } else {
				Level = Energy switch
				{
					  var n when n >= 1500 => 6,
					  var n when n >= 1000 => 5,
					  var n when n >= 600  => 4,
					  var n when n >= 300  => 3,
					  var n when n >= 200  => 2,
					  _ => 1
				};
				radius = Level - 0.5f;
			}
```

Reply:
christ yes please i wanted to use a switch here but i didn't know this was a thing

Reply:
Expression switches baby!